srcCpp/RTIDDSImpl.cxx

```diff
@@ -3850,6 +3990,45 @@ double RTIDDSImpl_FlatData<T>::obtain_dds_deserialize_time_cost_override(
 
     return (end-start) / (float) iters;
 }
+
+template <typename T>
+bool RTIDDSImpl_FlatData<T>::get_serialize_overhead_size(
+        unsigned int &overhead_size)
+{
+    typedef typename rti::flat::flat_type_traits<T>::builder Builder;
+    typedef typename rti::flat::PrimitiveSequenceBuilder<unsigned char>
+            BinDataBuilder;
+
+    unsigned long int serializedSize = 68 + RTI_FLATDATA_MAX_SIZE;
+    unsigned char *buffer = new unsigned char[serializedSize];
+
+    Builder builder(buffer, serializedSize);
+
+    // Leave uninitialized
+    builder.add_key();
+    builder.add_entity_id(0);
+    builder.add_seq_num(0);
+    builder.add_timestamp_sec(0);
+    builder.add_timestamp_usec(0);
+    builder.add_latency_ping(0);
+
+    /* Custom type does not have a bin_data field */
+    #ifndef RTI_CUSTOM_TYPE
+        // Add payload
+        BinDataBuilder bin_data = builder.build_bin_data();
+        bin_data.add_n(0);
+        bin_data.finish();
+    #endif
+
+    /*
+     * Sample size does not count RTI_XCDR_ENCAPSULATION_SIZE, we do not need to
```

Comment:
Why?

Reply:
There are two functions available, `buffer_size` and `sample_size`. 
`buffer_size` is equal to `sample_size() + RTI_XCDR_ENCAPSULATION_SIZE;`
But `sample_size` does not count the size of the encapsulation.